packages/miniflare/test/index.spec.ts

```diff
@@ -492,9 +510,152 @@ test("Miniflare: Accepts https requests", async (t) => {
       }
     }`,
   });
+  t.teardown(() => mf.dispose());
 
   const res = await mf.dispatchFetch("https://localhost");
   t.true(res.ok);
+  await res.arrayBuffer();
 
   t.assert(log.logs[0][1].startsWith("Ready on https://"));
 });
+
+queuesTest("Miniflare: getBindings() returns all bindings", async (t) => {
+  const tmp = await useTmp(t);
+  const blobPath = path.join(tmp, "blob.txt");
+  await fs.writeFile(blobPath, "blob");
+  const mf = new Miniflare({
+    modules: true,
+    script: `
+    export class DurableObject {}
+    export default { fetch() { return new Response(null, { status: 404 }); } }
+    `,
+    bindings: { STRING: "hello", OBJECT: { a: 1, b: { c: 2 } } },
+    textBlobBindings: { TEXT: blobPath },
+    dataBlobBindings: { DATA: blobPath },
+    serviceBindings: { SELF: "" },
+    d1Databases: ["DB"],
+    durableObjects: { DO: "DurableObject" },
+    kvNamespaces: ["KV"],
+    queueProducers: ["QUEUE"],
+    r2Buckets: ["BUCKET"],
+  });
+  t.teardown(() => mf.dispose());
+  const bindings = await mf.getBindings();
+
+  t.like(bindings, {
+    STRING: "hello",
+    OBJECT: { a: 1, b: { c: 2 } },
+    TEXT: "blob",
+  });
+  t.deepEqual(bindings.DATA, viewToBuffer(utf8Encode("blob")));
+
+  const opts: util.InspectOptions = { colors: false };
+  t.regex(util.inspect(bindings.SELF, opts), /name: 'Fetcher'/);
+  t.regex(util.inspect(bindings.DB, opts), /name: 'D1Database'/);
+  t.regex(util.inspect(bindings.DO, opts), /name: 'DurableObjectNamespace'/);
+  t.regex(util.inspect(bindings.KV, opts), /name: 'KvNamespace'/);
+  t.regex(util.inspect(bindings.QUEUE, opts), /name: 'WorkerQueue'/);
+  t.regex(util.inspect(bindings.BUCKET, opts), /name: 'R2Bucket'/);
+
+  // Check with WebAssembly binding (aren't supported by modules workers)
+  // (base64 encoded module containing a single `add(i32, i32): i32` export)
+  const addWasmModule =
+    "AGFzbQEAAAABBwFgAn9/AX8DAgEABwcBA2FkZAAACgkBBwAgACABagsACgRuYW1lAgMBAAA=";
+  const addWasmPath = path.join(tmp, "add.wasm");
+  await fs.writeFile(addWasmPath, Buffer.from(addWasmModule, "base64"));
+  await mf.setOptions({
+    script:
+      'addEventListener("fetch", (event) => event.respondWith(new Response(null, { status: 404 })));',
+    wasmBindings: { ADD: addWasmPath },
+  });
+  const { ADD } = await mf.getBindings<{ ADD: WebAssembly.Module }>();
+  const instance = new WebAssembly.Instance(ADD);
+  assert(typeof instance.exports.add === "function");
+  t.is(instance.exports.add(1, 2), 3);
```

Comment:
It would be good to see some verification of the poisoning logic after the `setOptions` call